src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -398,6 +417,9 @@ public override string GetDataTypeName(int ordinal)
 			case ColumnType.NewDecimal:
 				return "DECIMAL";
 
+			case ColumnType.Json:
+			    return "JSON";
```

Comment:
I want to support this (very soon) but haven't developed the full test suite yet.